include/zephyr/net/ieee802154_radio_openthread.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief This file extends interface of ieee802154_radio.h for OpenThread.
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_IEEE802154_RADIO_OPENTHREAD_H_
+#define ZEPHYR_INCLUDE_NET_IEEE802154_RADIO_OPENTHREAD_H_
+
+#include <zephyr/net/ieee802154_radio.h>
+
+/**
+ *  OpenThread specific capabilities of ieee802154 driver.
+ *  This type extends @ref ieee802154_hw_caps.
+ */
+enum ieee802154_openthread_hw_caps {
+	/** Capability to transmit with @ref IEEE802154_OPENTHREAD_TX_MODE_TXTIME_MULTIPLE_CCA
+	 *  mode.
+	 */
+	IEEE802154_OPENTHREAD_HW_MULTIPLE_CCA = BIT(IEEE802154_HW_CAPS_BITS_PRIV_START),
+};
+
+enum ieee802154_openthread_tx_mode {
+	/**
+	 * The @ref IEEE802154_OPENTHREAD_TX_MODE_TXTIME_MULTIPLE_CCA mode allows to send
+	 * a scheduled packet if the channel is reported idle after at most
+	 * 1 + max_extra_cca_attempts CCAs performed back-to-back.
+	 *
+	 * This mode allows transmission of a packet within a certain time window,
+	 * see the Thread 1.x.x specification, section xxx (TODO: add reference when published).
```

Comment:
This line is misleading. As the doc for the file specifies the features added here are to be part of OpenThread, but that doesn't imply that they must be part of Thread Specification. This extension should allow to add any proprietary features on top of Thread Specification minimum requirements.

I'd suggest to just remove this line. The feature itself is quite well explained here so there is no need to refer to any external specification.

Reply:
@edmont Robert had similar feedback. He proposed on Discord that we could simply call this an "experimental OpenThread feature".

So I'd propose wording similar to:

"This mode is a non-standard experimental OpenThread feature. It allows transmission of a packet within a certain time window."

Reply:
Changed the sentence to Florian's proposal.